### PR TITLE
[DEVHAS-235] Trim generated route CR name

### PR DESCRIPTION
**Pull request by johnmcollier**
Signed-off-by: John Collier <jcollier@redhat.com>

### What does this PR do?:
This PR updates the Route generation logic to ensure the CR name is no more than 30 characters. To ensure that the resultant generated hostname is valid.

### Which issue(s)/story(ies) does this PR fixes:
https://issues.redhat.com/browse/DEVHAS-235

### PR acceptance criteria:

- [ ] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: pkg/generate.go [hunk 1]
```diff
@@ -370,7 +377,7 @@ func generateRoute(options gitopsv1alpha1.GeneratorOptions) *routev1.Route {
 			APIVersion: "route.openshift.io/v1",
 		},
 		ObjectMeta: v1.ObjectMeta{
-			Name:      options.Name,
+			Name:      routeName,
```
Comment (keithchong) [inline pkg/generate.go:380]: Is this the only place that needs to be changed?  See also line ~390 or ~393

Also do we need to consider the other resources, eg the names of the other deployment, deployment patch and service resources?

Comment (johnmcollier) [reply pkg/generate.go:380]: No, this is specific to the route's metadata.name, as the generated host name is derived from it. The other resources are fine with the default name.

**Comment by johnmcollier** [comment]
@keithchong Just following up to see if you had any further questions/concerns around this

**Review by keithchong — APPROVED** [review]
Hi John, sorry about the delay.  Thanks for answering my question earlier.

LGTM

**Pull request merged**